pkg/shed/schema.go

```diff
@@ -147,17 +147,22 @@ func (db *DB) schemaIndexPrefix(name string) (id byte, err error) {
 // getSchema retrieves the complete schema from
 // the database.
 func (db *DB) getSchema() (s schema, err error) {
+	if db.sch != nil {
```

Comment:
since there's no mutex here, this will result in a panic

Reply:
How will testing if the schema is nil cause a panic?  We're not de-referencing db.sch here, but simply testing if it is populated.

Reply:
a pointer is still a memory address and has a value, that value is stored in the memory address. the fact that we are not dereferencing the value does not mean we are not accessing the same memory address (in this case of the pointer) and this needs synchronization. afaik when the runtime tries to read and write at the same time to the same memory location it will panic. the fact that you haven't seen the panic does not mean it will not happen. in any case, when accessing a shared resource across different goroutines it is accustomed to use a mutex in order to prevent data races